src/Persistence/ProxyGenerator.php

```diff
@@ -43,6 +46,19 @@ public static function wrap(object $object): Proxy
         return self::generateClassFor($object)::createLazyProxy(static fn() => $object); // @phpstan-ignore-line
     }
 
+    /**
+     * @template T of object
+     *
+     * @param PersistentProxyObjectFactory<T> $factory
+     * @phpstan-param Attributes $attributes
+     *
+     * @return T&Proxy<T>
+     */
+    public static function wrapFactory(PersistentProxyObjectFactory $factory, callable|array $attributes): Proxy
+    {
+        return self::generateClassFor($factory)::createLazyProxy(static fn() => unproxy($factory->create($attributes))); // @phpstan-ignore-line
```

Comment:
I'm using "our" proxy, but I think I'll directly use `ProxyHelper::generateLazyProxy()`. This way I think I'll find a way to open this feature to all kind of factories (and not only proxy ones)